### ISSUE TITLE
[architecture] which log libary to use

### ISSUE THREAD

**Issue by NissesSenap** [grafana-operator/grafana-operator-experimental#15]
Today we are using logrus which is something that flux also uses.

At the same time I'm wondering if it could be a good idea to go over to the official k8s log solution, klog:
https://kubernetes.io/docs/concepts/cluster-administration/system-logs/
https://pkg.go.dev/k8s.io/klog/v2

This isn't a issue to be implemented, more a issue to think of the best long-term solution for the operator.
What is other operators like 
* loki-operator
* kubebuilder in general
* flux
* etc etc

In short good to think about this before releasing v5.
